contractcourt/channel_arbitrator.go

```diff
@@ -509,6 +509,29 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 	// commitment transaction. We enter this state either due to an outside
 	// sub-system, or because an on-chain action has been triggered.
 	case StateBroadcastCommit:
+		// Under normal operation, we can only enter
+		// StateBroadcastCommit via a user or chain trigger. On restart,
+		// this state may be reexecuted after closing the channel, but
+		// failing to commit to StateContractClosed or
+		// StateFullyResolved. In that case, one of the three close
+		// triggers will be presented, signifying that we should skip
+		// rebroadcasting, and straight to resolving the on-chain
+		// contract or marking the channel resolved.
+		switch trigger {
+		case localCloseTrigger, remoteCloseTrigger:
+			nextState = StateContractClosed
+		case coopCloseTrigger:
+			nextState = StateFullyResolved
+		}
+
+		if nextState != StateDefault {
```

Comment:
This is a bit confusing, why not return directly from the switch?

Reply:
Fixed